contracts/cw3-flex-multisig/src/msg.rs

```diff
@@ -1,19 +1,137 @@
 use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
 
-use cosmwasm_std::{CosmosMsg, Empty, HumanAddr};
+use crate::error::ContractError;
+use cosmwasm_std::{CosmosMsg, Decimal, Empty, HumanAddr};
 use cw0::{Duration, Expiration};
-use cw3::Vote;
+use cw3::{ThresholdResponse, Vote};
 use cw4::MemberChangedHookMsg;
 
 #[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
 pub struct InitMsg {
     // this is the group contract that contains the member list
     pub group_addr: HumanAddr,
-    pub required_weight: u64,
+    pub threshold: Threshold,
     pub max_voting_period: Duration,
 }
 
+/// This defines the different ways tallies can happen.
+///
+/// The total_weight used for calculating success as well as the weights of each
+/// individual voter used in tallying should be snapshotted at the beginning of
+/// the block at which the proposal starts (this is likely the responsibility of a
+/// correct cw4 implementation).
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
+#[serde(rename_all = "snake_case")]
+pub enum Threshold {
```

Comment:
I've just noticed a there's duplicity between the descriptions in here, versus the ones in `ThresholdResponse` (part of `packages/cw3`).

Not wrong per se, but maybe a good idea to unify / centralize the docs in the spec. We can alsway use "See also" or simialr  to point to the main docs, by example.

Reply:
It is not used in `packages/cw3` explicitly, but rather when configuring an Init message in an implementation.

I guess we could move it there as a standard type to configure in `init`. I wanted to make sure the docs were visible, and if there was some way to link/import them that would be better

Reply:
I'll do it. And unify the docs.